app/models/submission.rb

```diff
@@ -482,6 +498,268 @@ def turnitinable_by_lti?
     turnitin_data.select{|_, v| v.is_a?(Hash) && v.key?(:outcome_response)}.any?
   end
 
+  # VeriCite
+
+  # this function will check if the score needs to be updated and update/save the new score if so,
+  # otherwise, it just returns the vericite_data_hash
+  def vericite_data()
+    self.vericite_data_hash ||= {}
+    self.vericite_data_hash[:provider] = :vericite
+    # check to see if the score is stale, if so, fetch it again
+    updateScores = false
+    # since there could be multiple versions, let's not waste calls for old versions and use the old score
+    if !self.readonly? && self.versions.current.number == self.version_number
+      self.vericite_data_hash.keys.each do |asset_string|
+        data = self.vericite_data_hash[asset_string]
+        next unless data && data.is_a?(Hash) && data[:object_id]
+        updateScores = updateScores || vericite_recheck_score(data)
+      end
+      # we have found at least one score that is stale, call VeriCite and save the results
+      if updateScores
+        check_vericite_status
+      end
```

Comment:
Its a combination of "vericite_recheck_score", "check_vericite_status", and "vericite.generateReport" 

The walkthrough looks like:

1) code asks for vericite data like: submission.vericite_data
2) This calls the function vericite_data()
3) vericite_data determines whether this data was recently checked by calling "vericite_recheck_score". If so, it just returns the cached value
4) If not, it calls "check_vericite_status"
5) "check_vericite_status" has extra checks to make sure it only calls the VeriCite API every once in a while as well. If these checks say to call, then it calls vericite.generateReport
6) "vericite.generateReport" calls "sendRequest" where "command == :get_scores". In this function, it has another check for cached data. The requests calls VeriCite for all scores for that particular user. Then stores the results in a map based on each file. 

I will look into making step 6 even more generic and look up all scores for the entire course in one call. This would mean that at most, only one call per assignment can be made to VeriCite's API every 5 minutes, not matter how many users are loading that assignment.